### ISSUE TITLE
Dashboard: error message running dashboard

### ISSUE THREAD

**Issue by hansthen** [clustervision/trinity#212]
**version: exeter** 

Dashboard does not work without a manual procedure.

```
2015-05-05 06:31:28,769 47485 INFO openstack_auth.forms Login successful for user "admin".
2015-05-05 06:33:52,003 47485 ERROR django.request Internal Server Error: /dashboard/admin/hpc_overview/
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python2.7/site-packages/horizon/decorators.py", line 36, in dec
    return view_func(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/horizon/decorators.py", line 84, in dec
    return view_func(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/horizon/decorators.py", line 52, in dec
    return view_func(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/horizon/decorators.py", line 36, in dec
    return view_func(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/django/views/generic/base.py", line 69, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/django/views/generic/base.py", line 87, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/horizon/tables/views.py", line 157, in get
    handled = self.construct_tables()
  File "/usr/lib/python2.7/site-packages/horizon/tables/views.py", line 140, in construct_tables
    tables = self.get_tables().values()
  File "/usr/lib/python2.7/site-packages/horizon/tables/views.py", line 196, in get_tables
    self._tables[self.table_class._meta.name] = self.get_table()
  File "/usr/lib/python2.7/site-packages/horizon/tables/views.py", line 207, in get_table
    self.table = self.table_class(self.request, **self.kwargs)
  File "/usr/lib/python2.7/site-packages/horizon/tables/base.py", line 1143, in __init__
    self._populate_data_cache()
  File "/usr/lib/python2.7/site-packages/horizon/tables/base.py", line 1227, in _populate_data_cache
    for column in self.get_columns():
  File "/usr/share/openstack-dashboard/openstack_dashboard/wsgi/../../openstack_dashboard/dashboards/admin/hpc_overview/tables.py", line 51, in get_columns
    hardwares=trinity.hardwares_list(self.request)
  File "/usr/share/openstack-dashboard/openstack_dashboard/wsgi/../../openstack_dashboard/api/trinity.py", line 49, in hardwares_list
    data=c.hardwares_list()
```

**Comment by hansthen** [comment]
I noticed trinity_api is supposed to be listening on port 32123, but it is not.

**Comment by abhimukh** [comment]
@hansthen 
What steps did you take leading to this error?
What is the output of the trinity_api log file?
Which manual procedure did you follow?

This can be due to 
1. The xCAT server is refusing requests from Trinity
  a. Because cv_install_xcat_ws was not run in the controller
  b. Because the apache conf file on the controller was not modified appropriately
2. The steps discussed in 1. were performed after running cv_install_api on the openstack node, and the Trinity API server was not (re)started.

**Comment by hansthen** [comment]
Probably because of 1a and 1b. Does this mean it is a duplicate?

**Comment by abhimukh** [comment]
If it is due to 1(a) and/or (b) then it is not a bug, but a case of procedures not being followed step by step. Should be closed.

**Comment by hansthen** [comment]
Why assign this to me?

**Comment by abhimukh** [comment]
@themisa Could you please verify that cv_install_xcat_ws is a postbootscript for the controller installation
[root@controller postscripts]# lsdef -t osimage -o centos7.0-x86_64-install-controller
Object name: centos7.0-x86_64-install-controller
    imagetype=linux
    osarch=x86_64
    osdistroname=centos7.0-x86_64
    osname=Linux
    osvers=centos7.0
    postbootscripts=cv_install_slapd,cv_install_xcat,cv_install_docker,__cv_install_xcat_ws__
    profile=controller
    provmethod=install

And, also, that the fix for #215 (1c0abe6cf7b02cabfddff46e78aa46e29b397763 ) works.

**Comment by themisa** [comment]
for *r3 0d12eb5
 (restored tables) cv_install_xcat_ws is a postbootscript
[root@controller ~]# lsdef -t osimage -o centos7.0-x86_64-install-controller
Object name: centos7.0-x86_64-install-controller
    imagetype=linux
    osarch=x86_64
    osdistroname=centos7.0-x86_64
    osname=Linux
    osvers=centos7.0
    postbootscripts=cv_install_slapd,cv_install_xcat,cv_install_docker,cv_install_xcat_ws
    profile=controller
    provmethod=install

I will close #215 from that issue
